api/v1beta1/watcher_webhook.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var watcherlog = logf.Log.WithName("watcher-resource")
+
+func (r *Watcher) SetupWebhookWithManager(mgr ctrl.Manager) error {
```

Comment:
nit: this is not really part of the API module

its a method that should be defined on the reconcile base class

https://github.com/openstack-k8s-operators/watcher-operator/blob/main/controllers/watcher_common.go#L81-L92

or factored out into a common class like this

https://github.com/openstack-k8s-operators/nova-operator/blob/main/api/v1beta1/common_webhook.go

there is actually still some technical debt in the nova operator to meve this from being defined on each webhook to webhook to a common subclass so we could proceed with this as is i just want to call out the fac that this will largely be identical for every webhooks so we should try and duplicate this in the futrure

Reply:
Thank you @SeanMooney , I have move this method into common_webhook.go